### PR TITLE
Wrd object access refactor

### PR DESCRIPTION
**Pull request by reiddraper**
See commit comment for details

### CODE REVIEW

File: src/riak_cs_wm_utils.erl [hunk 1]
```diff
@@ -544,56 +532,133 @@ object_access_authorize_helper(AccessType, Deletable,
...
+-spec translate_bucket_policy(atom(), binary(), pid()) ->
+    policy() | undefined.
+translate_bucket_policy(PolicyMod, Bucket, RiakPid) ->
+    case PolicyMod:bucket_policy(Bucket, RiakPid) of
+        {ok, P} ->
+            P;
+        {error, policy_undefined} ->
+            undefined;
+         {error, _Reason} ->
+             %% riak_cs_s3_response:api_error(Reason, RD, Ctx),
```
Comment (kellymclaughlin) [inline src/riak_cs_wm_utils.erl]: This was one of the things motivating my suggestion to refactor this. The only other error that should be returned is `multiple_bucket_owners` and I think we should fail the request on that.

File: src/riak_cs_wm_utils.erl [hunk 2]
```diff
@@ -515,26 +518,11 @@ object_access_authorize_helper(AccessType, Deletable,
...
-
-    Policy = PolicyMod:bucket_policy(Bucket, RiakPid),
+    #key_context{bucket=Bucket, manifest=Manifest} = LocalCtx,
+    CanonicalId = extract_canonical_id(User),
+    RequestedAccess = requested_access_helper(AccessType, Method),
+    ObjectAcl = extract_object_acl(Manifest),
+    Policy = translate_bucket_policy(PolicyMod, Bucket, RiakPid),
     Access = PolicyMod:reqdata_to_access(RD, AccessType, CanonicalId),
 
     case {riak_cs_acl:object_access(Bucket,
```
Comment (kellymclaughlin) [inline src/riak_cs_wm_utils.erl:540]: Any reason not to take another step and refactor this case into a function? It was one of the more complex parts of the bucket helper refactor so I can understand if that's the reasoning, but just wanted to get your thoughts.

**Comment by slfritchie** [comment]
The obnoxious `{error,function_clause,[{riak_cs_s3_policy,eval,[{access_v1,'PUT',.....` error has disappeared.  The refactoring look nice, no horrible dialyzer warnings added.  +1 to merge, thanks!

**Comment by kellymclaughlin** [comment]
+1 from me also

**Pull request merged**